graylog2-server/src/main/java/org/graylog/plugins/views/search/elasticsearch/QueryStringDecorators.java

```diff
@@ -18,29 +18,28 @@
 
 import org.graylog.plugins.views.search.ParameterProvider;
 import org.graylog.plugins.views.search.Query;
+import org.graylog.plugins.views.search.engine.PositionTrackingQuery;
 import org.graylog.plugins.views.search.engine.QueryStringDecorator;
 
 import javax.inject.Inject;
-import java.util.Collections;
 import java.util.Set;
 
-public class QueryStringDecorators implements QueryStringDecorator {
+public class QueryStringDecorators {
     private final Set<QueryStringDecorator> queryDecorators;
 
-    public static class Fake extends QueryStringDecorators {
-        public Fake() {
-            super(Collections.emptySet());
-        }
-    }
-
     @Inject
     public QueryStringDecorators(Set<QueryStringDecorator> queryDecorators) {
         this.queryDecorators = queryDecorators;
     }
 
-    @Override
     public String decorate(String queryString, ParameterProvider job, Query query) {
-        return this.queryDecorators.isEmpty() ? queryString : this.queryDecorators.stream()
-                .reduce(queryString, (prev, decorator) -> decorator.decorate(prev, job, query), String::concat);
+        return decorateWithPositions(queryString, job, query).getInterpolatedQuery();
+    }
+
+    public PositionTrackingQuery decorateWithPositions(String queryString, ParameterProvider job, Query query) {
+        return this.queryDecorators.stream()
+                .findFirst()
```

Comment:
I am just wondering... Why do we have a set of query decorators, while we always just pick the first one and ignore all the others?

Reply:
Good question - originally this was implemented in more generic way that allowed more decorators and combined their outputs. This would be a headache with the query positioning. Now we only have zero or one implementations (when enterprise), but we still need some dynamic way to link the decorator here in the server.

Reply:
So maybe we should now replace `Set<QueryStringDecorator>` with just `QueryStringDecorator` or`Optional<QueryStringDecorator>`?

Reply:
Fixed, good idea, thanks!